lexicons/tools/ozone/set/upsertSet.json

```diff
@@ -0,0 +1,24 @@
+{
+  "lexicon": 1,
+  "id": "tools.ozone.set.upsertSet",
+  "defs": {
+    "main": {
+      "type": "procedure",
+      "description": "Create or update set metadata",
+      "input": {
+        "encoding": "application/json",
+        "schema": {
+          "type": "ref",
+          "ref": "tools.ozone.set.defs#set"
```

Comment:
Hi there! Do you have any particular reason to make it a reference? This is a historical moment where the ref is used in the input of the procedure. Typically it is an object. I do not see any reuse of `tools.ozone.set.defs#set`. Only as input... 

I noticed because my Python code generator started raising errors because I did not implement ref resolving in the input schema. Now I've implemented it, but does it worth it

cc @bnewbold, @devinivy

Reply:
good question! I'm not sure if we support it other codegen tooling yet, guess we will find out.

my initial thought is that it is fine? the ref semantics are that this is basically the same as copying the `tools.ozone.set.defs#set` in to this location, and the type of that schema is `object`.

having it be a ref is nice because then the schema only needs to be updated in one place. even if it is only used in this one place in this set of Lexicions, having it be in a defs means other external Lexicons can reuse the definition (which they would not be able to do if it was just the "schema" of a "procedure").

I'm not sure if I would feel the same way about trying to use a ref in, for example, query params. things are basically symmetrical between `input` and `output`, and we allow refs in `output`, so seems like they should be allowed here.